src/content/interop/js-interop/package-web.md

```diff
@@ -0,0 +1,307 @@
+core web library like `dart:svg`.
+
+## Migrating from `dart:html`
+
+Remove the `dart:html` import and replace it with `package:web/web.dart`:
+
+```dart
+import 'dart:html' as html; // Remove
+import 'package:web/web.dart' as web; // Add
+```
+
+Add `web` to the `dependencies` in your pubspec:
+
+```yaml
+dependencies:
+  web: ^0.5.0
+```
+
+The following sections cover some of the common migration issues
+from `dart:html` to `package:web`.
+
+For any other migration issues, check the [dart-lang/web][] repo and file an
+issue.
+
+### Renames
+
+Many of the symbols in `dart:html` were renamed from
+their original IDL declaration to align more with Dart style.
+For example, `appendChild` became `append`,
+`HTMLElement` became `HtmlElement`, etc.
+
+In contrast, to reduce confusion,
+`package:web` uses the original names from the IDL definitions.
+A `dart fix` is available to convert types that have been renamed
+between `dart:html` and `package:web`.
+
+After changing the import, any renamed objects will be new "undefined" errors.
+You can address these either:
+- From the CLI, by running `dart fix --dry-run`.
+- In your IDE, by selecting the `dart fix`: **Rename to '`package:web name`'**.
+
+{% comment %}
+Maybe a pic here of menu selection in IDE?
+TODO: Update this documentation to refer to symbols instead of just types once
+we have a dart fix for that.
+{% endcomment -%}
+
+The `dart fix` covers many of the common type renames.
+If you come across a `dart:html` type without a built-in fix, let us know by
+filing an [issue][].
+You can manually discover the `package:web` type name
+by looking up the class' `@Native` annotation.
+You can do this by either:
+
+- Ctrl or cmd clicking the name in the IDE and choosing **Go to Definition**.
+- Searching for the name in the [`dart:html` API docs][html]
+  and checking its page under *Annotations*.
+
+The `@Native` annotation tells the compiler to treat any JS object of that type
+as the class that it annotates.
+
+Similarly, if you find an API with the keyword `native` in `dart:html` that
+doesn't have an equivalent in `package:web`, check to see if there was a rename
+with the `@JSName` annotation.
+The annotation tells the compiler to use the value in the annotation instead of
+the member name, much like [`@JS`] does.
+`native` is an internal keyword that means the same as `external` in this
+context.
+
+### Type checks
+
+It's common for code that uses `dart:html` to utilize runtime checks like `is`.
+When used with a `dart:html` object, `is` and `as` verify that the object is
+the JS type within the `@Native` annotation.
+In contrast, all `package:web` types are reified to [`JSObject`][]. This means a
+runtime type check will result in different behavior between `dart:html` and
+`package:web` types.
+
+To be able to perform type checks, migrate any `dart:html` code
+using `is` type tests to use [interop methods][] like `instanceOfString`.
```

Comment:
To encourage the more convenient and typed API, should we illustrate `isA<T>()` instead?

Reply:
Added

Reply:
I think it's fine to add it here. In the JS interop docs, I mention it's not available until 3.4 and the `@Since` annotation should make that clear anyways.